File: java-engine/src/main/java/engine/algorithms/faultless_dfs_dispersion/models/FaultlessDfsDispersionRobot.java
package engine.algorithms.faultless_dfs_dispersion.models;

import engine.core.models.Robot;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = false)
public class FaultlessDfsDispersionRobot extends Robot {

    private Long onID;
    private int parent;
    private int child;
    private int lastUsedPort;

}

File: dispersion-app/src/app/models/algorithms/faultless-dfs-dispersion/FaultlessDfsDispersionRobot.ts
import { Robot } from '../../core/Robot';
import { RobotState } from '../../utils/RobotState';

export class FaultlessDfsDispersionRobot extends Robot {
  onID: number;
  parent: number;
  child: number;
  lastUsedPort: number;

  constructor(
    id: number,
    state: RobotState,
    onID: number,
    parent: number,
    child: number,
    lastUsedPort: number
  ) {
    super(id, state);
    this.onID = onID;
    this.parent = parent;
    this.child = child;
    this.lastUsedPort = lastUsedPort;
  }
}
